Allow http/https scheme in otlpexporter endpoint configuration

alanwest opened the pull request:
**Description:** otlpexporter endpoint configuration allows for an `http` or `https` scheme. The scheme is stripped off when instantiating the gRPC channel. A scheme of `https` indicates a secure channel. This aligns the collector with this hopefully soon-to-be-merged specification PR https://github.com/open-telemetry/opentelemetry-specification/pull/1729

**Link to tracking Issue:** #2539

**Documentation:** Updated readme

jpkrohling commented on exporter/otlpexporter/README.md:
```diff
@@ -15,7 +15,9 @@ The following settings are required:
 using the gRPC protocol. The valid syntax is described
-[here](https://github.com/grpc/grpc/blob/master/doc/naming.md)
+[here](https://github.com/grpc/grpc/blob/master/doc/naming.md).
+Including a scheme of `http` or `https` is allowed in the `endpoint`, but not required.
```
Is it recommended, though? As someone reading this, I'm not sure if I should or not include the scheme.

alanwest replied:
Agreed, this was confusing. My intent was to highlight that a scheme of http or https is allowed whereas it was not before. Though it's probably unnecessary since things are pretty well described in the linked gRPC documentation https://github.com/grpc/grpc/blob/master/doc/naming.md, so I removed this sentence.

tigrannajaryan commented on exporter/otlpexporter/README.md:
```diff
@@ -15,7 +15,9 @@ The following settings are required:
-[here](https://github.com/grpc/grpc/blob/master/doc/naming.md)
+[here](https://github.com/grpc/grpc/blob/master/doc/naming.md).
+Including a scheme of `http` or `https` is allowed in the `endpoint`, but not required.
+If a scheme of `https` is used then client transport security overriding the `insecure` setting.
```
Can you please clarify this sentence? I am not sure I understand it.

alanwest replied:
Oops, yes I've left out some words, so this sentence makes no sense! Fixed.

The idea is that if you use the scheme `https` then the `insecure` setting is ignored. This is inline with the outstanding spec PR https://github.com/open-telemetry/opentelemetry-specification/pull/1729

tigrannajaryan reviewed (commented):
@alanwest thanks for working on this. Let's merge after https://github.com/open-telemetry/opentelemetry-specification/pull/1729 is merged.

alanwest commented:
@tigrannajaryan I've added a test validating the exporter when the endpoint has an `http` scheme. I'd like to add a test using the `https` scheme as well. This would require the test suite to be able to set up its mock receiver configured with TLS.

I noted that the `testdata` directory contains a certificate but not the corresponding key. Would it be ok to add a cert/key pair in the `testdata` directory?

The code that sets up the mock receiver is here https://github.com/open-telemetry/opentelemetry-collector/blob/d1fced9688ba925dcf84f9854eed0ddc49069ef2/exporter/otlpexporter/otlp_test.go#L75-L79

I essentially need to do something like the following to run the test using TLS:

```go
sopts := []grpc.ServerOption{}
creds, _ := credentials.NewServerTLSFromFile(grpctestdata.Path("testdata/test_cert.pem"), grpctestdata.Path("testdata/test_cert_key.pem"))
sopts = append(sopts, grpc.Creds(creds))

rcv := &mockTracesReceiver{
	mockReceiver: mockReceiver{
		srv: grpc.NewServer(sopts...),
	},
}
```

tigrannajaryan commented:
> I noted that the `testdata` directory contains a certificate but not the corresponding key. Would it be ok to add a cert/key pair in the `testdata` directory?

Yes, I think it is ok.

bogdandrutu commented:
Please rebase

alanwest commented:
Fixed merge conflicts and added a test using the scheme https. I had to create a new certificate in order to generate the private key. I replaced the existing cert in the testdata directory since it did not seem that it was used anywhere. Is this all good?

tigrannajaryan reviewed (approved):
LGTM. Thank you @alanwest
